### DISCUSSION TITLE
Git authentication on code-server on JupyterHub

### DISCUSSION THREAD

**Discussion by ggydush-fn** [coder/code-server#4974] [Q&A]
I'm curious if anyone has had success with git authentication on code-server running on JupyterHub.

I'm using [jupyter-server-proxy](https://github.com/jupyterhub/jupyter-server-proxy) to run code-server, but when trying to do any git actions, I'm taken to the following URL which does not exists, and I get `404 : Not Found`.

URL to access code-server
`<jupyterhub_url>/user/ggydush/codeserver/`

URL the code-server uses when trying to git authenticate
`<jupyterhub_url>/hub/callback?vscode-requestId=<request_id>&vscode-scheme=code-oss&vscode-authority=vscode.github-authentication&vscode-path=%2Fdid-authenticate&code=<code>&state=<state>`

I can replace `<jupyterhub_url>/hub/` with `<jupyterhub_url>/user/ggydush/codeserver/` and I get closer (the following is displayed).
![image](https://user-images.githubusercontent.com/69013027/157569482-29f117ac-5fa1-40ca-a059-4053b376b94d.png)

**Comment by jsjoeio** [comment]
I'm not familiar with JupyterHub or [jupyter-server-proxy](https://github.com/jupyterhub/jupyter-server-proxy). Do you think this is a bug with code-server itself?

**Reply by ggydush-fn** [reply]
I believe this may have solved itself on 4.1.0. Will report back here if that is not the case!
